app/src/main/kotlin/com/wire/android/ui/home/conversationslist/model/SearchQuery.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.ui.home.conversationslist.model
+
+data class SearchQuery(val text: String, val fromArchive: Boolean)
```

Comment:
I don't like this `boolean`. For now it's enough, but later the app could also have "folders" and "contacts" later. So IMO `enum class` would be preferable here.

Reply:
I think that Boris has a point. Also implementation with let's say `sourceType` sealed type shouldn't be much more difficult as now we only have Archive and "normal" conversations.

Reply:
I added comment about it but I can implement it right now 😄